### PR TITLE
Append tenant domain before resolving the username

### PR DESCRIPTION
**Pull request by mpmadhavig**
This is important when the associated local user of the federated user is getting resolved.

Resolves https://github.com/wso2/product-is/issues/18129

### CODE REVIEW

**Review by pavinduLakshan — DISMISSED** [review]

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp [hunk 1]
```diff
@@ -105,6 +105,12 @@
         tenantDomain = srtenantDomain;
     }
 
+    if (username.contains(IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR) && tenantDomain != null) {
```
Comment (Thisara-Welmilla) [inline identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:108]: What are the scenarios that the tenant domain is null? If the tenant domain is carbon.super is it possible to tenant domain to be null.

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp [hunk 2]
```diff
@@ -105,6 +105,12 @@
...
+        if (username.split(IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR)[1] != tenantDomain) {
+            username = username + IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR + tenantDomain;
```
Comment (Thisara-Welmilla) [inline identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:110]: What happens if the tenant domain and the domain name of the email is same. eg: tenant domain abc.com and the email is user@abc.com.

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp (hunk 1 above)
Comment (mpmadhavig) [reply identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:108]: > What are the scenarios that the tenant domain is null? 

The null check was added to preserve the default behavior for the cases where the tenant domain is passed as null. For this particular case tenant domain is never null.

> If the tenant domain is carbon.super is it possible to tenant domain to be null.

In the tested JIT provisioning flow, when the tenant domain is carbon.super, the value will not be null.

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp (hunk 2 above)
Comment (mpmadhavig) [reply identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:110]: Fixes with a20562c8c8c55669fd36d827e4d16e6863b0c746

**Review by Thisara-Welmilla — DISMISSED** [review]

**Review by Thisara-Welmilla — APPROVED** [review]

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**